frontend/src/components/RequestSourceInputs.vue

```diff
@@ -242,6 +267,101 @@ const inputValues: WritableComputedRef<RequestSource> = computed({
   },
 });
 
+const computedRules = computed(() => {
+  const sourceChainRules = {
+    $autoDirty: true,
+    required,
+  };
+  const tokenRules = {
+    $autoDirty: true,
+    required,
+  };
+  const requestFeeLoadingRules = {
+    $autoDirty: true,
+    sameAs: sameAs(false),
+  };
+
+  if (selectedToken.value) {
+    const selectedTokenValue = selectedToken.value.value;
+    const selectedTokenDecimals = selectedTokenValue.decimals;
+    const selectedAmountRules = {
+      $autoDirty: true,
+      required: helpers.withMessage('Amount is required', required),
+      isMatchingDecimals: helpers.withMessage(
+        `Max. number of decimals: ${selectedTokenDecimals}`,
+        isMatchingDecimals(selectedTokenDecimals),
+      ),
+    };
+    if (selectedTokenAmount.value) {
+      const min = TokenAmount.parse(
+        ['0.', '0'.repeat(selectedTokenDecimals - 1), '1'].join(''),
+        selectedTokenValue,
+      );
+      const selectedTokenAmountRules = {
+        $autoDirty: true,
+        minValue: helpers.withMessage(`Amount must be a positive number`, minTokenAmount(min)),
+      };
+      const totalRequestTokenAmountRules = {
+        $autoDirty: true,
+        maxValue: helpers.withMessage('Insufficient funds', (value: TokenAmount) => {
+          if (!value || !selectedToken.value) return true;
+          // Due to reactivity issues `max` has to be initialized here
+          const max = new TokenAmount({
+            amount: balance.value.toString(),
+            token: selectedToken.value.value,
+          });
+          return maxTokenAmount(max)(value);
+        }),
+      };
+      return {
+        sourceChain: sourceChainRules,
```

Comment:
Same as in the RequestTargetInputs. I think it would be nice if we could have one rules object that gets conditionally modified and then just one return at the end of the function.